contrib/dracut/90zfs/module-setup.sh.in

```diff
@@ -20,59 +20,30 @@ depends() {
 
 installkernel() {
 	instmods zfs
-	instmods zcommon
-	instmods znvpair
-	instmods zavl
-	instmods zunicode
-	instmods zlua
-	instmods icp
-	instmods spl
-	instmods zlib_deflate
-	instmods zlib_inflate
 }
 
 install() {
-	inst_rules @udevruledir@/90-zfs.rules
-	inst_rules @udevruledir@/69-vdev.rules
-	inst_rules @udevruledir@/60-zvol.rules
-	dracut_install hostid
-	dracut_install grep
-	dracut_install @sbindir@/zgenhostid
-	dracut_install @sbindir@/zfs
-	dracut_install @sbindir@/zpool
-	# Workaround for https://github.com/openzfs/zfs/issues/4749 by
-	# ensuring libgcc_s.so(.1) is included
-	if ldd @sbindir@/zpool | grep -qF 'libgcc_s.so'; then
-		# Dracut will have already tracked and included it
-		:;
-	elif command -v gcc-config >/dev/null 2>&1; then
-		# On systems with gcc-config (Gentoo, Funtoo, etc.):
-		# Use the current profile to resolve the appropriate path
-		s="$(gcc-config -c)"
-		dracut_install "/usr/lib/gcc/${s%-*}/${s##*-}/libgcc_s.so"*
-	elif [ "$(echo /usr/lib/libgcc_s.so*)" != "/usr/lib/libgcc_s.so*" ]; then
-		# Try a simple path first
-		dracut_install /usr/lib/libgcc_s.so*
-	elif [ "$(echo /lib*/libgcc_s.so*)" != "/lib*/libgcc_s.so*" ]; then
-		# SUSE
-		dracut_install /lib*/libgcc_s.so*
-	else
-		# Fallback: Guess the path and include all matches
-		dracut_install /usr/lib*/gcc/**/libgcc_s.so*
-	fi
-	# shellcheck disable=SC2050
-	if [ @LIBFETCH_DYNAMIC@ -gt 0 ]; then
-		for d in $libdirs; do
-			[ -e "$d/@LIBFETCH_SONAME@" ] && dracut_install "$d/@LIBFETCH_SONAME@"
-		done
-	fi
-	dracut_install @mounthelperdir@/mount.zfs
-	dracut_install @udevdir@/vdev_id
-	dracut_install awk
-	dracut_install cut
-	dracut_install tr
-	dracut_install head
-	dracut_install @udevdir@/zvol_id
+	inst_rules \
+		@udevruledir@/90-zfs.rules \
+		@udevruledir@/69-vdev.rules \
+		@udevruledir@/60-zvol.rules
```

Comment:
At this point, since @nabijaczleweli has advised me to keep the paths, I am unsure of what to do. Keeping the paths doesn't hurt, so let's keep them.

I was trying to check whether one can query the exit status of `inst_rules` to add an `exit 1` to the script but it doesn't seem to be possible. I see you've done that here [in zfsbootmenu](https://github.com/zbm-dev/zfsbootmenu/blob/9a03eab2b75647170bdc383903735a694ecd0ed6/dracut/module-setup.sh#L50) but I tried using the same construct to load a nonexistent udev rule on my local setup and it did not work. I don't think the `inst_rules "${_rule}"` statement ever returns a non-zero value. If it cannot find a rule it logs it to the console and skips ahead. For example, changing this bit to:

```sh
for i in 90-zfs.rules 69-vdev.rules 60-zvol.rules 01-does-not-exist.rules; do
    if ! inst_rules /lib/udev/rules.d/$i; then
        dfatal "Failed to install udev rule: $i"
        exit 1
    fi
done
```
results in the output (I've skipped the bits which don't matter):

```
...
dracut: *** Including module: zfs ***
dracut: Skipping udev rule: /lib/udev/rules.d/01-does-not-exist.rules
dracut: *** Including module: rootfs-block ***
...
```

I'm not sure if there is a way to handle this without resorting to `dracut-install` directly. Unfortunately, there's no dependency resolution associated with that.